[ADD] llm_anthropic: Anthropic Claude provider for Odoo 18.0

crottolo opened the pull request:
## Summary

This PR adds the **llm_anthropic** module, implementing full Anthropic Claude provider integration for Odoo 18.0.

The module was missing from the 18.0 branch (only a minimal implementation existed in 16.0 without tool calling support).

### Features
- ✅ All Claude models (4.5, 4, 3.x series) with automatic discovery
- ✅ Tool/function calling with Anthropic-specific format
- ✅ Extended thinking support (Claude's reasoning mode)
- ✅ Streaming responses
- ✅ Multimodal (vision) capabilities
- ✅ Message merging for Anthropic API requirements

### Technical Details

Key differences from OpenAI handled:
- System messages as separate `system` parameter (not in messages array)
- Tool format: `{"name", "description", "input_schema"}`
- Tool results: `role="user"` with `type="tool_result"`
- Consecutive user messages merged automatically (Anthropic requirement)

### Files Added (656 lines)
- `llm_anthropic/__manifest__.py`
- `llm_anthropic/models/anthropic_provider.py` (348 lines)
- `llm_anthropic/models/mail_message.py`
- `llm_anthropic/data/llm_publisher.xml`
- `llm_anthropic/data/llm_provider.xml`
- `llm_anthropic/README.md`
- Static assets (logo, banner)

## Test Plan
- [x] Module installs successfully on Odoo 18.0
- [x] Ruff linting passes
- [x] Chat functionality tested manually with Claude API
- [x] Tool calling verified

## Contributors
- Crottolo <bo@fl1.cz> - Odoo 18.0 port

crottolo committed:
[ADD] llm_anthropic: Anthropic Claude provider for Odoo 18.0

Implements full Anthropic provider integration following the project's
dispatch pattern. This was missing from 18.0 branch (only existed in 16.0
with minimal implementation).

Features:
- All Claude models (4.5, 4, 3.x series) with automatic discovery
- Tool/function calling with Anthropic-specific format
- Extended thinking support (Claude's reasoning mode)
- Streaming responses
- Multimodal (vision) capabilities
- Message merging for Anthropic API requirements

Key technical details:
- System messages handled as separate parameter (Anthropic requirement)
- Tool format: {"name", "description", "input_schema"}
- Tool results: role="user" with type="tool_result"
- Consecutive user messages merged automatically

adar2378 commented:
@crottolo Thank you for your contribution. I will check and get back to you.

adar2378 commented on llm_anthropic/models/anthropic_provider.py:
```diff
@@ -0,0 +1,357 @@
+        normalized = []
+        for msg in prepend_messages:
+            if msg.get("role") == "system":
+                continue
```
I think we should not remove system message from here.. Because it would be needed here:
https://github.com/apexive/odoo-llm/pull/210/files#diff-1f0416e7bd45616f2ecf43c6214fe9085ac8ea2bf9a4adcf203ec8f02f5e2579R85

adar2378 reviewed (changes requested):
@crottolo  Very good consistent coding patterns. I think we can merge it, just need to fix the bug with system message.

crottolo committed:
[FIX] llm_anthropic: keep system messages in normalize_prepend_messages

System messages were incorrectly filtered out in normalize_prepend_messages,
but they are needed later in anthropic_chat() to extract the system prompt.
This fix keeps system messages in the list so they can be properly extracted
and passed to the Anthropic API as the separate system parameter.

adar2378 commented:
@crottolo Thank you very much
